dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -1,22 +1,127 @@
+    except Exception as general_error:
+        notify.data_engineering(message.cant_find_schema(pipeline_config, general_error))
+        raise
+    
+    # Make sure pipeline_config contains a "pipeline" key
+    try:
+        pipeline = pipeline_config["pipeline"]
+    except KeyError:
+        notify.data_engineering(message.expected_config_key_missing("Pipeline key not found in config", "pipeline", "dataset_ingress_v1"))
+        raise ValueError(message.expected_config_key_missing("Pipeline key not found in config", "pipeline", "dataset_ingress_v1"))
+    except Exception as general_error:
+        notify.data_engineering(message.unexpected_error(f"Failed to get pipeline from config", general_error))
+        raise
+    
+    # Check for the existence of a pipeline configuration file
+    try:
+        if not local_store.has_lone_file_matching("pipeline-config.json"):
+            notify.data_engineering(message.expected_local_file_missing("Pipeline config not found", "pipeline-config.json", "dataset_ingress_v1"))
+            raise ValueError(message.expected_local_file_missing("Pipeline config not found", "pipeline-config.json", "dataset_ingress_v1"))
+    except Exception as general_error:
+        notify.data_engineering(message.unexpected_error("Failed to check for pipeline config", general_error))
+        raise
+
+    # Extract the patterns for required files from the pipeline configuration
+    try:
+        required_file_patterns = matching.get_required_files_patterns(pipeline_config)
+    except Exception as general_error:
+        notify.data_engineering(message.cant_find_schema(pipeline_config, general_error))
+        raise
+
+    # Validate that required_file_patterns is a list of strings
+    try:
+        required_file_patterns = matching.get_required_files_patterns(pipeline_config)
+        if not isinstance(required_file_patterns, list) or not all(isinstance(pattern, str) for pattern in required_file_patterns):
+            raise ValueError("required_file_patterns must be a list of strings")
+    except Exception as general_error:
+        notify.data_engineering(message.cant_find_schema(pipeline_config, general_error))
+        raise
+
+    # Check for the existence of each required file
+    for required_file in required_file_patterns:
+        try:
+            if not local_store.has_lone_file_matching(required_file):
+                notify.data_engineering(message.expected_local_file_missing(f"Required file {required_file} not found", required_file, "dataset_ingress_v1"))
+                raise ValueError(message.expected_local_file_missing(f"Required file {required_file} not found", required_file, "dataset_ingress_v1"))
+        except Exception as general_error:
+            notify.data_engineering(message.unexpected_error(f"Failed to check for required file {required_file}", general_error))
+            raise
+    
+    # Validate that supplementary_distribution_patterns is a list of strings
+    try:
```

Comment:
You dont need to check this, thats what this function does and its unit tested. We've made sure elsewhere it didnt error when populating a variable so this isn't needed.